Source/Core/Core/Core.cpp

```diff
@@ -351,17 +341,8 @@ static void CpuThread()
   if (_CoreParameter.bFastmem)
     EMM::InstallExceptionHandler();  // Let's run under memory watch
 
-  if (!s_state_filename.empty())
-  {
-    // Needs to PauseAndLock the Core
-    // NOTE: EmuThread should have left us in State::Stepping so nothing will happen
-    //   until after the job is serviced.
-    QueueHostJob([] {
-      // Recheck in case Movie cleared it since.
-      if (!s_state_filename.empty())
-        ::State::LoadAs(s_state_filename);
-    });
-  }
+  if (savestate_path)
+    QueueHostJob([savestate_path] { ::State::LoadAs(*savestate_path); });
```

Comment:
`&savestate_path`, otherwise I think this will copy the string.

Reply:
That won't work. `&savestate_path` would give you a pointer to `std::optional`, but `LoadAs` takes a const reference to `std::string`.

Reply:
Hmm, are you sure? AFAICT, adding the `&` will just capture `savestate_path` by reference, not take its address and make `savestate_path` a pointer.

Reply:
Oh, you mean in the capture list :) Sorry, I thought you meant in the call to `LoadAs`.